connect-example/src/main/java/com/stripe/android/connect/example/MainActivity.kt

```diff
@@ -2,137 +2,52 @@ package com.stripe.android.connect.example
-                            imageVector = Icons.Default.Settings,
-                            contentDescription = stringResource(R.string.settings),
+                NavHost(navController = navController, startDestination = MainDestination.ComponentPicker) {
+                    composable(MainDestination.ComponentPicker) {
+                        ComponentPickerContent(
+                            viewModel = viewModel,
+                            openSettings = { navController.navigate(route = MainDestination.Settings) },
                         )
                     }
-                    IconButton(
-                        onClick = {
-                            coroutineScope.launch {
-                                if (!sheetState.isVisible) {
-                                    sheetType = SheetType.APPEARANCE
-                                    sheetState.show()
-                                } else {
-                                    sheetState.hide()
-                                }
-                            }
-                        }
-                    ) {
-                        Icon(
-                            imageVector = Icons.Default.MoreVert,
-                            contentDescription = stringResource(R.string.customize_appearance),
-                        )
-                    }
-                }
-            } ?: { },
-            modalSheetState = sheetState,
-            modalContent = (embeddedComponentAsync is Success).then {
-                {
-                    BackHandler(enabled = sheetState.isVisible) {
-                        coroutineScope.launch { sheetState.hide() }
-                    }
-                    when (sheetType) {
-                        SheetType.SETTINGS -> SettingsView(
-                            onDismiss = { coroutineScope.launch { sheetState.hide() } },
+                    composable(MainDestination.Settings) {
+                        val settingsViewModel = hiltViewModel<SettingsViewModel>()
+                        SettingsView(
+                            viewModel = settingsViewModel,
+                            onDismiss = { navController.safeNavigateUp() },
                             onReloadRequested = viewModel::reload,
                         )
-                        SheetType.APPEARANCE -> AppearanceView(
-                            onDismiss = { coroutineScope.launch { sheetState.hide() } },
-                        )
                     }
                 }
-            },
-        ) {
-            EmbeddedComponentManagerLoader(
-                embeddedComponentAsync = embeddedComponentAsync,
-                reload = viewModel::reload,
-            ) {
-                ComponentPickerList()
             }
         }
     }
+}
 
-    private enum class SheetType {
-        SETTINGS,
-        APPEARANCE,
-    }
+@Suppress("ConstPropertyName")
+private object MainDestination {
+    const val ComponentPicker = "ComponentPicker"
```

Comment:
👀 Ignoring lint and sticking with PascalCase so minimize thrash when we can use the [type-safe DSL](https://developer.android.com/guide/navigation/design/type-safety).